lib/zlib.js

```diff
@@ -363,6 +363,9 @@ function Zlib(opts, mode) {
     var error = new Error(message);
     error.errno = errno;
     error.code = exports.codes[errno];
+    // TODO: Can't defer the error emission to nextTick because both the sync
```

Comment:
@chrisdickinson any idea how this can be solved? There is a chance that I might have totally misunderstood the problem here.